school/module.py

```diff
@@ -0,0 +1,1433 @@
+        else:
+            return "-"
+
+    def _translate_obligation(ctx, obligation: str) -> str:
+        """Translate obligation from it's czech shortcut"""
+        if obligation == "V":
+            return _(ctx, "Optional")
+        elif obligation == "PV":
+            return _(ctx, "Compulsory - optional")
+        elif obligation == "P":
+            return _(ctx, "Compulsory")
+        elif obligation == "O":
+            return _(ctx, "Other")
+        elif obligation == "D":
+            return _(ctx, "Recommended")
+        elif obligation == "VO":
+            return _(ctx, "Optional - specialized")
+        else:
+            return "-"
+
+    async def _parse_subject_parameters(
+        self, ctx: commands.Context, parameters: str
+    ) -> Optional[argparse.Namespace]:
+        """Parse parameters for subject editing"""
+        parser = CommandParser()
+        parser.add_argument("--abbreviation", type=str, nargs="+", default=None)
+        parser.add_argument("--name", type=str, nargs="+", default=None)
+        parser.add_argument("--institute", type=str, nargs="+", default=None)
+        parser.add_argument("--semester", type=str, choices=SEMESTERS)
+        parser.add_argument("--guarantor", type=int, nargs="+", default=None)
+        parser.add_argument("--teachers-add", type=int, nargs="+", default=None)
+        parser.add_argument("--teachers-remove", type=int, nargs="+", default=None)
+        args = parser.parse_args(shlex.split(parameters))
+        if parser.error_message:
+            await ctx.reply(
+                _(ctx, "Error parsing arguments:")
+                + f"\n> `{parser.error_message.replace('`', '')}`"
+            )
+            return None
+
+        for kw in ["abbreviation", "name", "institute", "semester"]:
+            if getattr(args, kw) is not None:
+                setattr(args, kw, " ".join(getattr(args, kw)))
+
+        if getattr(args, "guarantor") is not None:
+            setattr(args, "guarantor", getattr(args, "guarantor")[0])
+
+        return args
+
+    async def _parse_program_parameters(
+        self, ctx: commands.Context, parameters: str
+    ) -> Optional[argparse.Namespace]:
+        """Parse parameters for program editing"""
+        parser = CommandParser()
+        parser.add_argument("--abbreviation", type=str, nargs="+", default=None)
+        parser.add_argument("--name", type=str, nargs="+", default=None)
+        parser.add_argument("--degree", type=str, nargs="+", default=None)
+        args = parser.parse_args(shlex.split(parameters))
+        if parser.error_message:
+            await ctx.reply(
+                _(ctx, "Error parsing arguments:")
+                + f"\n> `{parser.error_message.replace('`', '')}`"
+            )
+            return None
+
+        for kw in ["abbreviation", "name", "degree"]:
+            if getattr(args, kw) is not None:
+                setattr(args, kw, " ".join(getattr(args, kw)))
+
+        return args
```

Comment:
These don't need to exist if code handles the data update

Reply:
Well yes, but I'd like to have the option to manually edit the data in some special cases.